### PR TITLE
Implement MCF parser in python

### PR DESCRIPTION
**Pull request by pradh**
This is a port of the C++ code from google3 with similar tests.  Some features, as noted in TODOs, are missing.

### CODE REVIEW

File: kg_util/mcf_parser.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    return prop in ('typeOf', 'subClassOf', 'subPropertyOf', 'rangeIncludes',
+                    'domainIncludes', 'specializationOf')
+
+
+def _is_common_ref_property(prop):
+    return (_is_schema_ref_property(prop) or prop == 'location' or
+            prop == 'observedNode' or prop == 'containedInPlace' or
+            prop == 'containedIn' or prop == 'populationType' or
+            prop == 'measuredProperty' or prop == 'measurementDenominator' or
+            prop == 'populationGroup' or prop == 'constraintProperties' or
```
Comment (shifucun) [inline kg_util/mcf_parser.py:63]: also use "or prop in [...]" ?

File: kg_util/mcf_parser.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+def _is_global_ref(value):
+    return (value.startswith(_dcid_prefix) or value.startswith(_dcs_prefix) or
+            value.startswith(_schema_prefix))
+
+
+def _is_local_ref(value):
+    return value.startswith(_local_prefix)
+
+
+@dataclasses.dataclass
```
Comment (shifucun) [inline kg_util/mcf_parser.py:76]: good to know this!

**Review by shifucun — APPROVED** [review]
very clear and easy to understand! We should have a Go version :)

**Review by beets — APPROVED** [review]
thanks for adding this!

would be nice to have a README as well

File: kg_util/mcf_parser.py (hunk 1 above)
Comment (pradh) [reply kg_util/mcf_parser.py]: Done.

**Comment by pradh** [comment]
> would be nice to have a README as well

Added README

**Comment by pradh** [comment]
Thanks for the review!  Addressed comments.

**Review by shifucun — APPROVED** [review]

**Comment by danbri** [comment]
Fantastic, thanks! :)

**Pull request merged**